config/configmaps/install-log-regexes-configmap.yaml

```
@@ -297,6 +285,11 @@ data:
       - "failed to initialize the cluster: Cluster operator monitoring is still updating"
       installFailingReason: MonitoringOperatorStillUpdating
       installFailingMessage: Timeout waiting for the monitoring operator to become ready
+    - name: NoWorkerNodesReady
+      searchRegexStrings:
+      - "Got 0 worker nodes, 3 master nodes.*none are schedulable or ready for ingress pods"
```

I would suggest we keep the regex for the number of master nodes here. Even if OSD clusters are always three masters today, they may not be in the future; and you also don't want to miss this event if, say, one of the masters _also_ didn't come up.
```suggestion
      - "Got 0 worker nodes, \\d+ master nodes.*none are schedulable or ready for ingress pods"
```

---

Makes sense, I'll make this change. From what I understand if any of the masters fail to come up they'll be caught by `BootstrapFailed` (which gets picked up before this failure reason would be):

```
Failed to wait for bootstrapping to complete. This error usually happens when there is a problem with control plane hosts that prevents the control plane operators from creating the control plane.
```

So I _think_ this will only kick in if we ever go to >3 masters, but this covers bases in case I'm wrong about `BootstrapFailed` or if that error falls away in the future.